services/QuillLMS/app/lib/synthetic/generators/base.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module Synthetic
+  module Generators
+    class Base < ApplicationService
+
+      class NotImplementedError < StandardError; end
+
+      attr_reader :results_hash
+
+      # takes an array of unique strings, e.g. ['original string', 'sample 2']
+      def initialize(string_array, options = {})
+        # create and hash with each string  to an empty hash, e.g.
+        # {'string' => {}, 'string 2' => {}
+        @results_hash = string_array.map {|string| [string,{}] }.to_h
```

Comment:
arrays has a `to_h` which allows passing of a block: `string_array.to_h { |string| [string, {}] }`

Since this construction is already documented in the [docs](https://ruby-doc.org/core-2.7.0/Array.html#method-i-to_h),  you could leave out the the comments on line 13-14.

Reply:
Thanks for the tip, I didn't realize `to_h` could take a block. Change made!